containers/daap-python-base/src/var/task/data_product_metadata.py

```diff
@@ -1,116 +1,290 @@
+            "Parameters": {},
+            "StoredAsSubDirectories": False,
+        },
+        "PartitionKeys": [],
+        "TableType": "EXTERNAL_TABLE",
+        "Parameters": {"classification": "csv", "skip.header.line.count": "1"},
+    },
+}
 
-def get_data_product_metadata_spec_path(version: str = "") -> str:
-    # if version is empty we'll get the latest version
-    if version == "":
-        file_paths = get_filepaths_from_s3_folder(
-            DataProductConfig.metadata_spec_prefix().uri
-        )
+
+def get_data_product_specification_path(
+    spec_type: JsonSchemaName, version: None | str = None
+) -> str:
+    """gets path for given version or latest version of metadata and schema json schema"""
+
+    # if version is empty we get the latest version
+    if version is None:
+        file_paths = get_filepaths_from_s3_folder(specification_prefix(spec_type).uri)
         versions = list(
             {i for p in file_paths for i in p.split("/") if i.startswith("v")}
         )
         versions.sort(key=lambda x: [int(y.replace("v", "")) for y in x.split(".")])
         latest_version = versions[-1]
-        path = DataProductConfig.metadata_spec_path(latest_version)
+        path = specification_path(spec_type, latest_version)
     else:
-        DataProductConfig.metadata_spec_path(version)
-        path = DataProductConfig.metadata_spec_path(version)
+        path = specification_path(spec_type, version)
 
     return path.uri
 
 
-def split_bucket_and_key(path):
-    bucket, key = path.replace("s3://", "").split("/", 1)
-    return bucket, key
-
-
-class DataProductMetadata:
-    """
-    class to handle creation and updating of
-    metadata relating to a dataproduct
-    """
+class BaseJsonSchema:
+    """base class for operations on json type metadata and schema for data products"""
 
-    def __init__(self, data_product_name: str, logger: DataPlatformLogger):
+    def __init__(
+        self,
+        data_product_name: str,
+        logger: DataPlatformLogger,
+        json_type: JsonSchemaName,
+        write_bucket_path: BucketPath,
+    ):
         self.data_product_name = data_product_name
         self.logger = logger
-        bucket, key = split_bucket_and_key(
-            DataProductConfig(name=data_product_name).metadata_path().uri
-        )
-        self.metadata_bucket = bucket
-        self.metadata_key = key
-        self._check_if_metadata_exists()
+        self.valid = False
+        self.type = json_type
+        self.exists = self._check_exists()
+        self.write_bucket = write_bucket_path.bucket
+        self.latest_version_key = write_bucket_path.key
+        if not self.exists:
+            self.write_key = write_bucket_path.key
+        self.version = write_bucket_path.key.split("/")[1]
 
-    def _check_if_metadata_exists(self) -> object:
+    def _check_exists(self) -> object:
+        if self.type == JsonSchemaName.data_product_schema:
+            bucket_path = BucketPath.from_uri(
```

Comment:
is this different from the `write_key` / `write_bucket`? I was thinking it would be the same thing

Reply:
yeah is different because when this is called from the eventual update endpoints the `write_bucket` will be the same as the latest version path, but the `write_key` will not as they'll need to be some logic applied to infer the increment type and the path for updated metadata and schema. So actually the path set at __init__ will only be a `write_key` for new data products metadata or schema, otherwise it's a latest version key and `write_key` should be set elsewhere after the version increment type is known.

`_check_exists` also now only checks if the instantiated object exists by just checking if there's a `v1.0`